From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sat, 7 Jun 2025 19:48:12 +0800
Subject: [PATCH] Rebrand to LemonMint


diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index e529f9c42fe9362400fb16c8b7e66e860844abba..45be16ad8cb4c9fd497620aa0691ef904c94c5d5 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/dev.bacteriawa.mint/mint-api/pom.properties"); // Mint
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/me.coderfrish.lemint/lemint-api/pom.properties"); // LemonMint
         Properties properties = new Properties();
 
         if (stream != null) {

diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
index d3d3d6468c09b5f33ac98f6938b849d413d1af3d..2675f693b6408fbf41bcc78b977bf0b94ce5ba04 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -31,7 +31,7 @@ public record ServerBuildInfoImpl(
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";

     private static final String BRAND_PAPER_NAME = "Paper";
-    private static final String BRAND_MINT_NAME = "Mint"; // Mint
+    private static final String BRAND_LEMINT_NAME = "LemonMint"; // Mint

     private static final String BUILD_DEV = "DEV";

@@ -43,9 +43,9 @@ public record ServerBuildInfoImpl(
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(BRAND_MINT_ID), // Mint
+                .orElse(BRAND_LEMINT_ID), // Mint
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse(BRAND_MINT_NAME), // Mint
+                .orElse(BRAND_LEMINT_NAME), // Mint
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
