From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Fri, 1 Aug 2025 22:19:19 +0800
Subject: [PATCH] Added Player Join Event for new plugin system.


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 2f7dad299b0076457a5ecf8fbb036d760fbfb4e1..512208231ca4584971e325766617ce01594c9eb4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -312,10 +312,13 @@ public class CraftPlayer extends CraftHumanEntity implements Player, PluginMessa
     private net.kyori.adventure.text.Component playerListFooter; // Paper - Adventure
     private @Nullable Set<net.kyori.adventure.bossbar.BossBar> activeBossBars;
 
+    public final me.coderfrish.player.Player lemonPlayer;// LemonMint - Lua Player API
+
     public CraftPlayer(CraftServer server, ServerPlayer entity) {
         super(server, entity);
 
         this.firstPlayed = System.currentTimeMillis();
+        this.lemonPlayer = new me.coderfrish.player.Player(this);
     }
 
     public static net.minecraft.world.entity.Relative deltaRelativeToNMS(io.papermc.paper.entity.TeleportFlag.Relative apiFlag) {
