From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sat, 2 Aug 2025 23:27:41 +0800
Subject: [PATCH] Fixed git rebase to LemonMint.


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
index 780e98534ae83cb3742dae6784aced479b1ed4c3..f4c16a5776caa0aece2b56b5e6e2d0569f37aa5b 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -31,7 +31,7 @@ public record ServerBuildInfoImpl(
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";
 
     private static final String BRAND_PAPER_NAME = "Paper";
-    private static final String BRAND_MINT_NAME = "LemonMint"; // Mint
+    private static final String BRAND_LEMONMINT_NAME = "LemonMint"; // Mint
 
     private static final String BUILD_DEV = "DEV";
 
@@ -43,9 +43,9 @@ public record ServerBuildInfoImpl(
         this(
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_ID)
                 .map(Key::key)
-                .orElse(BRAND_LEMONMINT_ID), // Mint
+                .orElse(BRAND_LEMONMINT_ID), // LemonMint
             getManifestAttribute(manifest, ATTRIBUTE_BRAND_NAME)
-                .orElse(BRAND_MINT_NAME), // Mint
+                .orElse(BRAND_LEMONMINT_NAME), // LemonMint
             SharedConstants.getCurrentVersion().id(),
             SharedConstants.getCurrentVersion().name(),
             getManifestAttribute(manifest, ATTRIBUTE_BUILD_NUMBER)
