From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Mon, 4 Aug 2025 18:26:28 +0800
Subject: [PATCH] Fixed Plugin Loaded Error.


diff --git a/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java b/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
index 10c2def089b99c263302c432c97276c2b3f354d1..cddb63aacb186bb4041edf6f1734acb8a8b1d50b 100644
--- a/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
+++ b/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
@@ -19,7 +19,6 @@ import io.papermc.paper.plugin.provider.type.spigot.SpigotPluginProvider;
 import java.lang.reflect.Type;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Set;
 import java.util.TreeMap;
 
 import me.coderfrish.plugin.LuaPluginManager;
