From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sat, 9 Aug 2025 00:22:36 +0800
Subject: [PATCH] Fixed Global entity error of that swipe the screen.


diff --git a/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java b/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
index e417c9a140778954572fb8bc190fad64feacffe0..538db016c3d139fd762b3c849f9301bb612e4a20 100644
--- a/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
+++ b/src/main/java/ca/spottedleaf/moonrise/common/util/TickThread.java
@@ -262,7 +262,7 @@ public class TickThread extends Thread {
                     }
                     return true;
                 } else {
-                    return false;
+                    return true;
                 }
             }
             if (isShutdownThread()) {
