From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Wed, 30 Jul 2025 22:39:04 +0800
Subject: [PATCH] Easily fixed bukkit scheduler api


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index a4519b295174642979c34fe220a8b884b2c6ec0d..1c65118a3e6703ffc3480a8a9f85adee3141449d 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1069,7 +1069,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         if (this.server != null) {
             this.server.spark.disable(); // Paper - spark
             this.server.disablePlugins();
-            this.server.waitForAsyncTasksShutdown(); // Paper - Wait for Async Tasks during shutdown
+            if (!me.coderfrish.config.experiment.BukkitSchedulerConfig.enabled) this.server.waitForAsyncTasksShutdown(); // Paper - Wait for Async Tasks during shutdown
         }
         // CraftBukkit end
         if (io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper != null) io.papermc.paper.plugin.PluginInitializerManager.instance().pluginRemapper.shutdown(); // Paper - Plugin remapping
@@ -1267,7 +1267,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             // Spigot start
             // Paper start
             LOGGER.info("Running delayed init tasks");
-            this.server.getScheduler().mainThreadHeartbeat(); // run all 1 tick delay tasks during init,
+            if (!me.coderfrish.config.experiment.BukkitSchedulerConfig.enabled) this.server.getScheduler().mainThreadHeartbeat(); // run all 1 tick delay tasks during init,
             // this is going to be the first thing the tick process does anyways, so move done and run it after
             // everything is init before watchdog tick.
             // anything at 3+ won't be caught here but also will trip watchdog....
@@ -1600,7 +1600,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
                     this.autoSave();
                 }
 
-                this.server.getScheduler().mainThreadHeartbeat(); // CraftBukkit
+                if(!me.coderfrish.config.experiment.BukkitSchedulerConfig.enabled) this.server.getScheduler().mainThreadHeartbeat(); // CraftBukkit
                 // Paper start - avoid issues with certain tasks not processing during sleep
                 Runnable task;
                 while ((task = this.processQueue.poll()) != null) {
