From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Tue, 22 Jul 2025 19:31:11 +0800
Subject: [PATCH] Fix bukkit api of scoreboard.


diff --git a/net/minecraft/commands/Commands.java b/net/minecraft/commands/Commands.java
index 3be546ea638dbde77196bbc2dc8a5b6a87e308c4..9c6839a493cd39a934a4b7fecf540e427ea584d5 100644
--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -225,7 +225,8 @@ public class Commands {
         //RotateCommand.register(this.dispatcher); // Folia - region threading - TODO later
         SayCommand.register(this.dispatcher);
         //ScheduleCommand.register(this.dispatcher); // Folia - region threading
-        //ScoreboardCommand.register(this.dispatcher, context); // Folia - region threading
+        if (me.coderfrish.config.experiment.ScoreboardConfig.enabled)
+            ScoreboardCommand.register(this.dispatcher, context); // Folia - region threading
         SeedCommand.register(this.dispatcher, selection != Commands.CommandSelection.INTEGRATED);
         VersionCommand.register(this.dispatcher, selection != Commands.CommandSelection.INTEGRATED);
         SetBlockCommand.register(this.dispatcher, context);
