From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Mon, 25 Aug 2025 17:02:58 +0800
Subject: [PATCH] Rebrand to LemonMint.


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 03ee024d4f592177c2d0c134dfb5959352f42074..9ef9c3b77fe2f38e48c499f05a061fcf1685ba43 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -26,6 +26,13 @@ public interface ServerBuildInfo {
     Key BRAND_MINT_ID = Key.key("menthamc", "mint");
     // Mint end
 
+    /* LemonMint start */
+    /**
+     * The brand id for Mint.
+     */
+    Key BRAND_LEMON_MINT_ID = Key.key("menthamc", "lemonmint");
+    /* LemonMint end */
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
