From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 14 Sep 2025 10:11:59 +0800
Subject: [PATCH] Disabled `folia-supported` field in plugin.yml or
 paper-plugin.yml.


diff --git a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
index de5d5399e7f9cf56439ada32f3cc7df3a2e95ef4..7034507abca2bacf4e9924ecdd0f2205f1144d94 100644
--- a/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
+++ b/src/main/java/org/bukkit/plugin/PluginDescriptionFile.java
@@ -269,7 +269,6 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
     // Paper end - plugin loader api
     // Folia start - block plugins not marked as supported
     private String foliaSupported;
-    private static final String FOLIA_SUPPORTED_KEY = "folia-supported";
 
     /**
      * Returns whether the plugin has been marked to be compatible with regionised threading as provided
@@ -277,7 +276,7 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
      */
     @Override
     public boolean isFoliaSupported() {
-        return this.foliaSupported != null && this.foliaSupported.equalsIgnoreCase("true");
+        return true;
     }
     // Folia end - block plugins not marked as supported
     // Paper start - oh my goddddd
@@ -1281,9 +1280,9 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
             prefix = map.get("prefix").toString();
         }
         // Folia start - block plugins not marked as supported
-        if (map.get(FOLIA_SUPPORTED_KEY) != null) {
-            foliaSupported = map.get(FOLIA_SUPPORTED_KEY).toString();
-        }
+//        if (map.get(FOLIA_SUPPORTED_KEY) != null) {
+//            foliaSupported = map.get(FOLIA_SUPPORTED_KEY).toString();
+//        }
         // Folia end - block plugins not marked as supported
     }
 
@@ -1362,9 +1361,9 @@ public final class PluginDescriptionFile implements io.papermc.paper.plugin.conf
             map.put("prefix", prefix);
         }
         // Folia start - block plugins not marked as supported
-        if (foliaSupported != null) {
-            map.put(FOLIA_SUPPORTED_KEY, foliaSupported);
-        }
+//        if (foliaSupported != null) {
+//            map.put(FOLIA_SUPPORTED_KEY, foliaSupported);
+//        }
         // Folia end - block plugins not marked as supported
 
         return map;
