From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Fri, 3 Oct 2025 02:07:50 +0800
Subject: [PATCH] Rebased Folia to Traium.


diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index fd3553bdc1c3cdbf6aa3dc00e0a4987f8eaa4fb8..423f14dfb1a70db26f120d37d7f077bd4ba83f12 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -109,6 +109,8 @@ public class Main {
                 JvmProfiler.INSTANCE.start(Environment.SERVER);
             }
 
+            me.coderfrish.traium.config.TraiumConfig.loadAllConfig(); // Traium
+
             io.papermc.paper.plugin.PluginInitializerManager.load(optionSet); // Paper
             Bootstrap.bootStrap();
             Bootstrap.validate();
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 196996aa8094bd53b61f2797b4ff3b9bcae16132..88afd5a855fab41698217a2000375067962d338a 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1968,6 +1968,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
+        if (true) return me.coderfrish.traium.config.GlobalConfig.serverBrandName;
         return io.papermc.paper.ServerBuildInfo.buildInfo().brandName(); // Paper
     }
 
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index d8bab0ba6e2374fb74f3fa9f7db802f442240414..952bae0ab74dc849bd643317555e7502507e6dd3 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -171,6 +171,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         this.paperConfigurations.initializeGlobalConfiguration(this.registryAccess());
         this.paperConfigurations.initializeWorldDefaultsConfiguration(this.registryAccess());
         // Paper end - initialize global and world-defaults configuration
+        me.coderfrish.traium.config.TraiumConfig.setupTraium();
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
diff --git a/net/minecraft/server/packs/repository/ServerPacksSource.java b/net/minecraft/server/packs/repository/ServerPacksSource.java
index 3d482f3eed2d320477d5375a04ebdde22820a00f..c082c92f319804b9b9c648c0873514e50722c081 100644
--- a/net/minecraft/server/packs/repository/ServerPacksSource.java
+++ b/net/minecraft/server/packs/repository/ServerPacksSource.java
@@ -71,9 +71,12 @@ public class ServerPacksSource extends BuiltInPackSource {
         // Paper start - custom built-in pack
         final PackLocationInfo info;
         final PackSelectionConfig packConfig;
-        if ("paper".equals(id)) {
+        if ("paper".equals(id) || "traium".equals(id)) {
             info = new PackLocationInfo(id, title, PackSource.BUILT_IN, Optional.empty());
             packConfig = new PackSelectionConfig(true, Pack.Position.TOP, true);
+        } else if("bukkit".equals(id)) {
+            info = new PackLocationInfo("file/bukkit", title, PackSource.WORLD, Optional.empty());
+            packConfig = new PackSelectionConfig(true, Pack.Position.TOP, true);
         } else {
             info = createBuiltInPackLocation(id, title);
             packConfig = FEATURE_SELECTION_CONFIG;
