From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Mon, 6 Oct 2025 00:14:14 +0800
Subject: [PATCH] Restored Function And Vanilla Command.


diff --git a/io/papermc/paper/threadedregions/RegionizedServer.java b/io/papermc/paper/threadedregions/RegionizedServer.java
index e1842ab8f20271a8b20ab8462b09a4745a612a9a..b1e5e0546166d42855847d73de2dfccdffe7cfe9 100644
--- a/io/papermc/paper/threadedregions/RegionizedServer.java
+++ b/io/papermc/paper/threadedregions/RegionizedServer.java
@@ -322,6 +322,9 @@ public final class RegionizedServer {
             this.globalTick(world, tickCount);
         }
 
+        // tick functions
+        MinecraftServer.getServer().getFunctions().tick();
+
         // tick connections
         this.tickConnections();
 
@@ -417,6 +420,10 @@ public final class RegionizedServer {
         // weather cycle
         this.advanceWeatherCycle(world);
 
+        // ScheduledEvents Tick
+        long l = world.levelData.getGameTime() + 1L;
+        world.serverLevelData.getScheduledEvents().tick(world.getServer(), l);
+
         // sleep status
         this.checkNightSkip(world);
 
diff --git a/net/minecraft/commands/Commands.java b/net/minecraft/commands/Commands.java
index ad19e9e599b26dfdb53929d9410860cb1b97698c..aebd72b1b8b4591993c6a0ee1e46bea46125ef3e 100644
--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -189,8 +189,8 @@ public class Commands {
         ClearInventoryCommands.register(this.dispatcher, context);
         //CloneCommands.register(this.dispatcher, context); // Folia - region threading - TODO
         DamageCommand.register(this.dispatcher, context);
-        //DataCommands.register(this.dispatcher); // Folia - region threading - TODO
-        //DataPackCommand.register(this.dispatcher, context); // Folia - region threading - TODO
+//        DataCommands.register(this.dispatcher); // Folia - region threading - TODO
+        DataPackCommand.register(this.dispatcher, context); // Folia - region threading - TODO
         //DebugCommand.register(this.dispatcher); // Folia - region threading - TODO
         DefaultGameModeCommands.register(this.dispatcher);
         //DialogCommand.register(this.dispatcher, context); // Folia - region threading - TODO
@@ -202,7 +202,7 @@ public class Commands {
         FillCommand.register(this.dispatcher, context);
         FillBiomeCommand.register(this.dispatcher, context);
         ForceLoadCommand.register(this.dispatcher);
-        //FunctionCommand.register(this.dispatcher); // Folia - region threading - TODO
+        FunctionCommand.register(this.dispatcher); // Folia - region threading - TODO
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher, context);
         GiveCommand.register(this.dispatcher, context);
@@ -224,7 +224,7 @@ public class Commands {
         //RideCommand.register(this.dispatcher); // Folia - region threading - TODO later
         //RotateCommand.register(this.dispatcher); // Folia - region threading - TODO later
         SayCommand.register(this.dispatcher);
-        //ScheduleCommand.register(this.dispatcher); // Folia - region threading
+        ScheduleCommand.register(this.dispatcher); // Folia - region threading
         //ScoreboardCommand.register(this.dispatcher, context); // Folia - region threading
         SeedCommand.register(this.dispatcher, selection != Commands.CommandSelection.INTEGRATED);
         VersionCommand.register(this.dispatcher, selection != Commands.CommandSelection.INTEGRATED);
@@ -235,7 +235,7 @@ public class Commands {
         //SpreadPlayersCommand.register(this.dispatcher); // Folia - region threading - TODO later
         StopSoundCommand.register(this.dispatcher);
         SummonCommand.register(this.dispatcher, context);
-        //TagCommand.register(this.dispatcher); // Folia - region threading - TODO later
+        TagCommand.register(this.dispatcher); // Folia - region threading - TODO later
         //TeamCommand.register(this.dispatcher, context); // Folia - region threading - TODO later
         //TeamMsgCommand.register(this.dispatcher); // Folia - region threading - TODO later
         TeleportCommand.register(this.dispatcher);
diff --git a/net/minecraft/server/Main.java b/net/minecraft/server/Main.java
index 423f14dfb1a70db26f120d37d7f077bd4ba83f12..c0bae3bc2d8a47ef23a502578afee9dbe048e9cb 100644
--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -234,13 +234,14 @@ public class Main {
 
             PackRepository packRepository = ServerPacksSource.createPackRepository(levelStorageAccess);
             // CraftBukkit start
-            File bukkitDataPackFolder = new File(levelStorageAccess.getLevelPath(net.minecraft.world.level.storage.LevelResource.DATAPACK_DIR).toFile(), "bukkit");
-            if (!bukkitDataPackFolder.exists()) {
-                bukkitDataPackFolder.mkdirs();
-            }
-            File mcMeta = new File(bukkitDataPackFolder, "pack.mcmeta");
-            try {
-                com.google.common.io.Files.asCharSink(mcMeta, java.nio.charset.StandardCharsets.UTF_8).write("""
+            if (false) {
+                File bukkitDataPackFolder = new File(levelStorageAccess.getLevelPath(net.minecraft.world.level.storage.LevelResource.DATAPACK_DIR).toFile(), "bukkit");
+                if (!bukkitDataPackFolder.exists()) {
+                    bukkitDataPackFolder.mkdirs();
+                }
+                File mcMeta = new File(bukkitDataPackFolder, "pack.mcmeta");
+                try {
+                    com.google.common.io.Files.asCharSink(mcMeta, java.nio.charset.StandardCharsets.UTF_8).write("""
                     {
                         "pack": {
                             "description": "Data pack for resources provided by Bukkit plugins",
@@ -248,9 +249,10 @@ public class Main {
                         }
                     }
                     """.formatted(SharedConstants.getCurrentVersion().packVersion(net.minecraft.server.packs.PackType.SERVER_DATA))
-                );
-            } catch (java.io.IOException ex) {
-                throw new RuntimeException("Could not initialize Bukkit datapack", ex);
+                    );
+                } catch (java.io.IOException ex) {
+                    throw new RuntimeException("Could not initialize Bukkit datapack", ex);
+                }
             }
             java.util.concurrent.atomic.AtomicReference<WorldLoader.DataLoadContext> worldLoader = new java.util.concurrent.atomic.AtomicReference<>();
             // CraftBukkit end
