From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Tue, 14 Oct 2025 13:06:11 +0800
Subject: [PATCH] Restore Vanilla Datapack And Function.


diff --git a/io/papermc/paper/threadedregions/RegionizedServer.java b/io/papermc/paper/threadedregions/RegionizedServer.java
index 8c0971eb48c64ba902d20c65ec7b4db93dd1d142..a0fa8d81afec228f20b13081e14a6df2ecf1b446 100644
--- a/io/papermc/paper/threadedregions/RegionizedServer.java
+++ b/io/papermc/paper/threadedregions/RegionizedServer.java
@@ -319,6 +319,9 @@ public final class RegionizedServer {
         // tick player ping sample
         this.tickPlayerSample();
 
+        // tick functions
+        MinecraftServer.getServer().getFunctions().tick();
+
         // tick worlds
         for (final ServerLevel world : this.worlds) {
             this.globalTick(world, tickCount);
@@ -419,6 +422,9 @@ public final class RegionizedServer {
         // weather cycle
         this.advanceWeatherCycle(world);
 
+        long l = world.levelData.getGameTime() + 1L;
+        world.serverLevelData.getScheduledEvents().tick(world.getServer(), l);
+
         // sleep status
         this.checkNightSkip(world);
 
diff --git a/net/minecraft/commands/Commands.java b/net/minecraft/commands/Commands.java
index ad19e9e599b26dfdb53929d9410860cb1b97698c..7263ab318cd48bd088f6db9be0cf192a9363350d 100644
--- a/net/minecraft/commands/Commands.java
+++ b/net/minecraft/commands/Commands.java
@@ -190,7 +190,7 @@ public class Commands {
         //CloneCommands.register(this.dispatcher, context); // Folia - region threading - TODO
         DamageCommand.register(this.dispatcher, context);
         //DataCommands.register(this.dispatcher); // Folia - region threading - TODO
-        //DataPackCommand.register(this.dispatcher, context); // Folia - region threading - TODO
+        DataPackCommand.register(this.dispatcher, context); // Folia - region threading - TODO
         //DebugCommand.register(this.dispatcher); // Folia - region threading - TODO
         DefaultGameModeCommands.register(this.dispatcher);
         //DialogCommand.register(this.dispatcher, context); // Folia - region threading - TODO
@@ -202,7 +202,7 @@ public class Commands {
         FillCommand.register(this.dispatcher, context);
         FillBiomeCommand.register(this.dispatcher, context);
         ForceLoadCommand.register(this.dispatcher);
-        //FunctionCommand.register(this.dispatcher); // Folia - region threading - TODO
+        FunctionCommand.register(this.dispatcher); // Folia - region threading - TODO
         GameModeCommand.register(this.dispatcher);
         GameRuleCommand.register(this.dispatcher, context);
         GiveCommand.register(this.dispatcher, context);
