From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Tue, 14 Oct 2025 14:08:41 +0800
Subject: [PATCH] Force fixed command block execution.


diff --git a/net/minecraft/world/level/BaseCommandBlock.java b/net/minecraft/world/level/BaseCommandBlock.java
index 8b8ce26de104211728d4b64e641a2b471ce89073..cca4417b445392d71f9dfb8de4d3a98097e04416 100644
--- a/net/minecraft/world/level/BaseCommandBlock.java
+++ b/net/minecraft/world/level/BaseCommandBlock.java
@@ -96,7 +96,7 @@ public abstract class BaseCommandBlock implements CommandSource {
     }
 
     public boolean performCommand(Level level) {
-        if (true) return false; // Folia - region threading
+        if (!me.coderfrish.traium.config.WorldConfig.forceCommandBlockExecution) return false;
         if (level.isClientSide || level.getGameTime() == this.lastExecution) {
             return false;
         } else if ("Searge".equalsIgnoreCase(this.command)) {
