From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Tue, 22 Jul 2025 19:31:19 +0800
Subject: [PATCH] Fix bukkit api of scoreboard.


diff --git a/src/main/java/dev/bacteriawa/mint/config/MintConfig.java b/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
index d1c3faee25afa2d035fd40f531b9705e323b8fe2..0e724317263ba2d64b3dba6bf5cb51cff79db0da 100644
--- a/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
+++ b/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
@@ -126,7 +126,7 @@ public class MintConfig {
     }
 
     private static Set<Class<?>> getClassesByPackage() {
-        try(ScanResult result = new ClassGraph().acceptPackages("dev.bacteriawa.mint.config.modules").scan()) {
+        try(ScanResult result = new ClassGraph().acceptPackages("dev.bacteriawa.mint.config.modules", "me.coderfrish.config").scan()) {
             return result.getAllClasses().stream().map(ClassInfo::loadClass).collect(Collectors.toSet());
         }
     }
diff --git a/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java b/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
index 395c8e2d80b8493b8b04b43d3b1f9b3c29930f7e..c77078c8b7c906da60b6a8dec41574bcd14aba06 100644
--- a/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
+++ b/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
@@ -7,5 +7,5 @@ import dev.bacteriawa.mint.config.Configuration;
 @Configuration(name = "server-brand-name", type = ConfigCategory.misc)
 public class ServerBrandNameConfig {
     @ConfigField
-    public static String serverModName= "Mint";
+    public static String serverModName= "LemonMint";
 }
