From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CoderFrish <1573880184@qq.com>
Date: Wed, 25 Feb 2026 23:51:04 +0800
Subject: [PATCH] Online Player Profile Archives.


diff --git a/net/minecraft/core/UUIDUtil.java b/net/minecraft/core/UUIDUtil.java
index fd08566668cb04b3826c2e90ebe578b9225c60f0..00cad6f50f94678087a264f2fe7f32b7392ed474 100644
--- a/net/minecraft/core/UUIDUtil.java
+++ b/net/minecraft/core/UUIDUtil.java
@@ -86,6 +86,9 @@ public final class UUIDUtil {
     }
 
     public static UUID createOfflinePlayerUUID(final String playerName) {
+        if (dev.coderfrish.lirael.profiles.OnlinePlayerProfile.has(playerName))
+            return dev.coderfrish.lirael.profiles.OnlinePlayerProfile.getUUID(playerName);
+
         return UUID.nameUUIDFromBytes(("OfflinePlayer:" + playerName).getBytes(StandardCharsets.UTF_8));
     }
 
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index a2dd5af4752e9801b19b43bcb76f40598301e92d..9113579a4ddefd1a8a2e44a55d2efad6d1909417 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -579,6 +579,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         try {
             dev.coderfrish.lirael.config.LiraelConfig.save(); /* lirael config */
+            dev.coderfrish.lirael.profiles.OnlinePlayerProfile.saveProfile(); /* online profile */
         } catch (Exception e) {
             throw new IllegalStateException(e);
         }
diff --git a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index d924369285878ff73811b8186cdb16e76f5dda90..1036b562cc4fab0ce4c3b0114cb8d358b398ca5f 100644
--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -334,6 +334,7 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
         String playerName = gameprofile.name();
         java.net.InetAddress address = ((java.net.InetSocketAddress) this.connection.getRemoteAddress()).getAddress();
         java.util.UUID uniqueId = gameprofile.id();
+        if (server.usesAuthentication()) dev.coderfrish.lirael.profiles.OnlinePlayerProfile.addPlayer(playerName, uniqueId);
         final org.bukkit.craftbukkit.CraftServer server = ServerLoginPacketListenerImpl.this.server.server;
 
         // Paper start - Add more fields to AsyncPlayerPreLoginEvent
