From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CoderFrish <1573880184@qq.com>
Date: Wed, 25 Feb 2026 23:51:04 +0800
Subject: [PATCH] Online Player Profile Archives.


diff --git a/net/minecraft/core/UUIDUtil.java b/net/minecraft/core/UUIDUtil.java
index e0df8dcc542a2c158955f37abf4d3dd63310868a..d24ee0d3b722f100a1cb691c47778bd31a07bc9a 100644
--- a/net/minecraft/core/UUIDUtil.java
+++ b/net/minecraft/core/UUIDUtil.java
@@ -86,6 +86,9 @@ public final class UUIDUtil {
     }
 
     public static UUID createOfflinePlayerUUID(String username) {
+        if (dev.coderfrish.lirael.profiles.OnlinePlayerProfile.has(username))
+            return dev.coderfrish.lirael.profiles.OnlinePlayerProfile.getUUID(username);
+
         return UUID.nameUUIDFromBytes(("OfflinePlayer:" + username).getBytes(StandardCharsets.UTF_8));
     }
 
diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 38de7a838443acc71d5ea8a231be9f6f33d9aae2..b526fc26d1d543b43c9340defb39c98a637b10ea 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -546,6 +546,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
 
         try {
             dev.coderfrish.lirael.config.LiraelConfig.save(); /* lirael config */
+            dev.coderfrish.lirael.profiles.OnlinePlayerProfile.saveProfile(); /* online profile */
         } catch (Exception e) {
             throw new IllegalStateException(e);
         }
diff --git a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
index 69f694995848173c2f3137aa8e9706be6fd3a9ae..f8f8247c6fb0f3be8601fcbd7ded69b9795593ac 100644
--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -324,6 +324,7 @@ public class ServerLoginPacketListenerImpl implements ServerLoginPacketListener,
         String playerName = gameprofile.name();
         java.net.InetAddress address = ((java.net.InetSocketAddress) this.connection.getRemoteAddress()).getAddress();
         java.util.UUID uniqueId = gameprofile.id();
+        if (server.usesAuthentication()) dev.coderfrish.lirael.profiles.OnlinePlayerProfile.addPlayer(playerName, uniqueId);
         final org.bukkit.craftbukkit.CraftServer server = ServerLoginPacketListenerImpl.this.server.server;
 
         // Paper start - Add more fields to AsyncPlayerPreLoginEvent
