--- a/mint-server/build.gradle.kts
+++ b/mint-server/build.gradle.kts
@@ -51,7 +_,22 @@
         }
     }
 
-    activeFork = mint
+    val lemonMint = forks.register("lemonMint") {
+        forks = mint
+        upstream.patchRepo("paperServer") {
+            upstreamRepo = mint.patchedRepo("paperServer")
+            patchesDir = rootDirectory.dir("lemonMint-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+        upstream.patchDir("mintServer") {
+            upstreamPath = "mint-server"
+            excludes = setOf("src/minecraft", "paper-patches", "minecraft-patches", "build.gradle.kts", "build.gradle.kts.patch")
+            patchesDir = rootDirectory.dir("lemonMint-server/mint-patches")
+            outputDir = rootDirectory.dir("mint-server")
+        }
+    }
+
+    activeFork = lemonMint
 
 
     spigot {
@@ -141,12 +_,16 @@
         resources { srcDir("../paper-server/src/main/resources") }
         java { srcDir("../folia-server/src/main/java") }
         resources { srcDir("../folia-server/src/main/resources") }
+        java { srcDir("../mint-server/src/main/java") }
+        resources { srcDir("../mint-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
         java { srcDir("../folia-server/src/test/java") }
         resources { srcDir("../folia-server/src/test/resources") }
+        java { srcDir("../mint-api/src/test/java") }
+        resources { srcDir("../mint-api/src/test/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -173,7 +_,7 @@
 }
 
 dependencies {
-    implementation(project(":mint-api"))
+    implementation(project(":lemonMint-api"))
     implementation("io.github.classgraph:classgraph:4.8.179") // Mint
     implementation("com.electronwill.night-config:toml:3.6.6") // Mint
     // Abomination start
@@ -259,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Mint",
+            "Implementation-Title" to "LemonMint",
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Mint",
+            "Specification-Title" to "LemonMint",
             "Specification-Version" to project.version,
             "Specification-Vendor" to "Mentha Team",
-            "Brand-Id" to "menthamc:mint",
-            "Brand-Name" to "Mint",
+            "Brand-Id" to "menthamc:lemonMint",
+            "Brand-Name" to "LemonMint",
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
