From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CoderFrish <1573880184@qq.com>
Date: Wed, 25 Feb 2026 14:10:39 +0800
Subject: [PATCH] Blank Lirael Configuration System.


diff --git a/src/main/java/io/papermc/paper/PaperBootstrap.java b/src/main/java/io/papermc/paper/PaperBootstrap.java
index d543b1b107ab8d3eeb1fc3c1cadf489928d2786e..39e9da5d514ef73d0b6b5e29bd51babcf0971376 100644
--- a/src/main/java/io/papermc/paper/PaperBootstrap.java
+++ b/src/main/java/io/papermc/paper/PaperBootstrap.java
@@ -1,6 +1,8 @@
 package io.papermc.paper;
 
 import java.util.List;
+
+import dev.coderfrish.lirael.bootstrap.LiraelBootstrap;
 import joptsimple.OptionSet;
 import net.minecraft.SharedConstants;
 import net.minecraft.server.Main;
@@ -13,11 +15,12 @@ public final class PaperBootstrap {
     private PaperBootstrap() {
     }
 
-    public static void boot(final OptionSet options) {
+    public static void boot(final OptionSet options) throws Exception {
         SharedConstants.tryDetectVersion();
 
         getStartupVersionMessages().forEach(LOGGER::info);
 
+        LiraelBootstrap.bootstrap(); /* lirael server bootstrap */
         Main.main(options);
     }
 
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 9d4d97fcb0f3cc94e9a263c61cc0e7e58fd322c3..0cde720d8a182de267f96db4e9932d450e1271b1 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -19,6 +19,11 @@ public class Main {
     // Paper start - Reset loggers after shutdown
     static {
         System.setProperty("java.util.logging.manager", "io.papermc.paper.log.CustomLogManager");
+
+        if (System.console() == null) {
+            /* see https://github.com/MenthaMC/Mint/pull/24 */
+            System.setProperty(net.minecrell.terminalconsole.TerminalConsoleAppender.JLINE_OVERRIDE_PROPERTY, "false");
+        }
     }
     // Paper end - Reset loggers after shutdown
 
