From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Wed, 6 Aug 2025 18:30:14 +0800
Subject: [PATCH] Setup JavaScript Plugin Manager.


diff --git a/src/main/java/org/bukkit/command/PluginCommand.java b/src/main/java/org/bukkit/command/PluginCommand.java
index 551c5af6a7bfa2268cbc63be8e70d129bccaa912..c86ae34dff8d8d1d83781dd479411cae1fe7eb2a 100644
--- a/src/main/java/org/bukkit/command/PluginCommand.java
+++ b/src/main/java/org/bukkit/command/PluginCommand.java
@@ -14,7 +14,7 @@ public final class PluginCommand extends Command implements PluginIdentifiableCo
     private CommandExecutor executor;
     private TabCompleter completer;
 
-    PluginCommand(@NotNull String name, @NotNull Plugin owner) {
+    public PluginCommand(@NotNull String name, @NotNull Plugin owner) {
         super(name);
         this.executor = owner;
         this.owningPlugin = owner;
