From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 10 Aug 2025 10:01:03 +0800
Subject: [PATCH] Fixed Bukkit Plugin used old scheduler thus leading to tick
 thread check error.


diff --git a/src/main/java/org/bukkit/scheduler/BukkitTask.java b/src/main/java/org/bukkit/scheduler/BukkitTask.java
index d0fdccd7f097bcb4733ec35767177aa075b96e9b..b87e2c71f6ff7de9a3b4a105af104003b8f062d8 100644
--- a/src/main/java/org/bukkit/scheduler/BukkitTask.java
+++ b/src/main/java/org/bukkit/scheduler/BukkitTask.java
@@ -42,11 +42,4 @@ public interface BukkitTask {
      * Will attempt to cancel this task.
      */
     public void cancel();
-
-    /**
-     * Get execution state of folia scheduler
-     *
-     * @return #{@return io.papermc.paper.threadedregions.scheduler.ScheduledTask.ExecutionState}
-     */
-    public ScheduledTask.ExecutionState getState();
 }
