From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sat, 12 Jul 2025 17:19:34 +0800
Subject: [PATCH] Added Structure seed safe config.


diff --git a/src/main/java/dev/bacteriawa/mint/config/MintConfig.java b/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
index d1c3faee25afa2d035fd40f531b9705e323b8fe2..78cb11fe94298a56aebcabdffdb3937251ba0f7d 100644
--- a/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
+++ b/src/main/java/dev/bacteriawa/mint/config/MintConfig.java
@@ -126,7 +126,7 @@ public class MintConfig {
     }
 
     private static Set<Class<?>> getClassesByPackage() {
-        try(ScanResult result = new ClassGraph().acceptPackages("dev.bacteriawa.mint.config.modules").scan()) {
+        try(ScanResult result = new ClassGraph().acceptPackages("dev.bacteriawa.mint.config.modules", "me.coderfrish.config.modules").scan()) {
             return result.getAllClasses().stream().map(ClassInfo::loadClass).collect(Collectors.toSet());
         }
     }
