From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 14 Sep 2025 10:11:59 +0800
Subject: [PATCH] Disabled `folia-supported` field in plugin.yml or
 paper-plugin.yml.


diff --git a/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java b/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
index fb7c6621e2805f4339c255f6c2e02c55ff4c502e..abc13ea19814544c77c6fdfaaec34e9a5f4284d3 100644
--- a/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
+++ b/src/main/java/io/papermc/paper/plugin/provider/configuration/PaperPluginMeta.java
@@ -64,7 +64,7 @@ public class PaperPluginMeta implements PluginMeta {
     private PermissionConfiguration permissionConfiguration = new PermissionConfiguration(PermissionDefault.OP, List.of());
     @Required
     private ApiVersion apiVersion;
-    private boolean foliaSupported = false; // Folia
+//    private boolean foliaSupported = false; // Folia
 
     private Map<PluginDependencyLifeCycle, Map<String, DependencyConfiguration>> dependencies = new EnumMap<>(PluginDependencyLifeCycle.class);
 
@@ -255,7 +255,7 @@ public class PaperPluginMeta implements PluginMeta {
     // Folia start
     @Override
     public boolean isFoliaSupported() {
-        return this.foliaSupported;
+        return true;
     }
     // Folia end
 
