From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Wed, 30 Jul 2025 22:39:04 +0800
Subject: [PATCH] Easily fixed bukkit scheduler api


diff --git a/src/main/java/org/bukkit/scheduler/BukkitScheduler.java b/src/main/java/org/bukkit/scheduler/BukkitScheduler.java
index 446c115f19c4dde0d5e269f8d120bc51ad3898e0..996a3868d30bdfcdff4cfdc4e3134b897ca44f41 100644
--- a/src/main/java/org/bukkit/scheduler/BukkitScheduler.java
+++ b/src/main/java/org/bukkit/scheduler/BukkitScheduler.java
@@ -204,6 +204,7 @@ public interface BukkitScheduler {
      *
      * @return Active workers
      */
+    @Deprecated
     @NotNull
     public List<BukkitWorker> getActiveWorkers();
 
@@ -213,6 +214,7 @@ public interface BukkitScheduler {
      *
      * @return Active workers
      */
+    @Deprecated
     @NotNull
     public List<BukkitTask> getPendingTasks();
 
@@ -474,7 +476,14 @@ public interface BukkitScheduler {
      * @param plugin the reference to the plugin scheduling tasks
      * @return an executor associated with the given plugin
      */
+    @Deprecated(since = "1.21.8")
     @NotNull
     public java.util.concurrent.Executor getMainThreadExecutor(@NotNull Plugin plugin);
     // Paper end
+
+    /**
+     * This method is designed to never block or wait for locks; an immediate execution of all current tasks.
+     */
+    @Deprecated
+    public void mainThreadHeartbeat();
 }
