From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Fri, 3 Oct 2025 20:09:58 +0800
Subject: [PATCH] Rebased Folia to Traium.


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 652ff54e7c50412503725d628bfe72ed03059790..304cfd16ae1ece2476eac43f8e98dc286152bde7 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -19,6 +19,11 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
+    /**
+     * The brand id for Traium.
+     */
+    Key BRAND_TRAIUM_ID = Key.key("traiummc", "traium");
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
