From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Frish2021 <1573880184@qq.com>
Date: Sun, 14 Sep 2025 13:27:39 +0800
Subject: [PATCH] Rebrand to LemonMint.


diff --git a/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java b/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
index 963221fc5f4119bb0df7dba48ece334718fbafbd..1da3f65289f53c685cef60efe16b1ffd4e5e6ed2 100644
--- a/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
+++ b/src/main/java/dev/bacteriawa/mint/config/modules/misc/ServerBrandNameConfig.java
@@ -7,5 +7,5 @@ import dev.bacteriawa.mint.config.annotation.ConfigField;
 @Config(name = "server_brand_name", category = ConfigCategory.misc)
 public class ServerBrandNameConfig {
     @ConfigField
-    public static String serverModName= "Mint";
+    public static String serverModName= "LemonMint";
 }
